mirdata/datasets/idmt_smt_audio_effects.py

```diff
@@ -0,0 +1,287 @@
+    """
+
+    def __init__(self, data_home=None, version="default"):
+        super().__init__(
+            data_home,
+            version,
+            name="idmt_smt_audio_effects",
+            track_class=Track,
+            bibtex=BIBTEX,
+            indexes=INDEXES,
+            remotes=REMOTES,
+            download_info=DOWNLOAD_INFO,
+            license_info=LICENSE_INFO,
+        )
+
+    @core.cached_property
+    def _metadata(self):
+        """Return a dictionary containing metadata information parsed from XML files.
+
+        Returns:
+            dict: A dictionary containing metadata information parsed from XML files.
+
+        Raises:
+            FileNotFoundError: If metadata file not found.
+            ValueError: If there's an error parsing the XML file.
+            Exception: For unexpected errors during processing.
+        """
+        metadata = dict()
+        metadata = {
+            "fileID": {
+                "list_id": str,
+                "instrument": str,
+                "midi_nr": str,
+                "fx_group": int,
+                "fx_type": int,
+                "fx_setting": int,
+            }
+        }
+
+        xml_files_count = 0
+
+        for root, dirs, files in os.walk(self.data_home):
+            for file in files:
+                if file.endswith(".xml"):
+                    xml_files_count += 1
+                    xml_path = os.path.join(root, file)
+                    try:
+                        with open(xml_path, "r") as fhandle:
+                            tree = ET.parse(fhandle)
+
+                    except ET.ParseError:
+                        raise ValueError(
+                            f"Error parsing XML file {xml_path}. The file may be corrupted."
+                        )
+
+                    root_xml = tree.getroot()
+                    listID = root_xml.find("listinformation/listID").text
+
+                    for audiofile in root_xml.findall("audiofile"):
+                        name = audiofile.find("fileID").text
+                        instrument = audiofile.find("instrument").text
+                        midinr = audiofile.find("midinr").text
+                        fxgroup = audiofile.find("fxgroup").text
+                        fxtype = audiofile.find("fxtype").text
+                        fxsetting = audiofile.find("fxsetting").text
+
+                        metadata[name] = {
+                            "list_id": listID,
+                            "instrument": instrument,
+                            "midi_nr": int(midinr),
+                            "fx_group": int(fxgroup),
+                            "fx_type": int(fxtype),
+                            "fx_setting": int(fxsetting),
+                        }
+
+        if xml_files_count == 0:
+            raise FileNotFoundError(
+                f"No XML files found in {self.data_home}. Did you run .download?"
+            )
+        return metadata
```

Comment:
Cool! Now that I think about it... maybe it'd be nice to check that ALL files are there? Well we are actually throwing this error above:
```
except ET.ParseError:
    raise ValueError(
        f"Error parsing XML file {xml_path}. The file may be corrupted."
    )
```
Maybe with saying here `The file may be corrupted or not available, make sure you have all files.`, it's enough... what do you think @guillemcortes @magdalenafuentes?

Reply:
Hi,

_(Minor comment)_
Since you are not using `dirs`, maybe I'll change `for root, dirs, files in os.walk(self.data_home):` for `for root, _, files in os.walk(self.data_home):`

Also, the ParseError is only raised if the xml cannot be parsed, but the file exists (os.walk finds it), so I think the message is appropiate. Maybe you can also add something like `The file may be corrupted or not available`.

To check that all the xml have been processed maybe you can add to the final `if xml_files_count == 0` an additional `else if` that checks that `xml_files_count==expected_xml_count` and raises an error if the numbers don't coincide. So you can check that the total amount of xml files that have been loaded is the same amount that is expected.